### DISCUSSION TITLE
Reraise exception in yield except block but get FastAPIError finally.

### DISCUSSION THREAD

**Discussion by seokhyuncheong1** [fastapi/fastapi#11408] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
async def redis_session() -> AsyncGenerator[Redis, None]:
    async with RedisPool() as client:
        try:
            yield client
        except Exception:
            logger.warning(f'redis_session error: {traceback.format_exc()}')
            raise
```

### Description

Hi, our teams are considering updating FastAPI to v0.110.1.
After update FastAPI to v0.110.1, we suffer due to FastAPIError.
we use `yield` object in logic, and reraise exception in except block like below code.
```python
async def redis_session() -> AsyncGenerator[Redis, None]:
    async with RedisPool() as client:
        try:
            yield client
        except Exception:
            logger.warning(f'redis_session error: {traceback.format_exc()}')
            raise
```
We use redis session with `Depends()`.
When we raise custom exception, fastapi return FastAPIError response and we can see below error msg.
```
No response object was returned. There's a high chance that the application code is raising an exception and a dependency with yield has a block with a bare except, or a block with except Exception, and is not raising the exception again. Read more about it in the docs:
https://fastapi.tiangolo.com/tutorial/dependencies/dependencies-with-yield/#dependencies-with-yield-and-except
```
We add custom_exception_handler to FastAPI and it is not working...
Also, we return `JSONResponse` in custom exception.
Please, give me some solutions...

### Operating System

macOS

### FastAPI Version

0.110.1

### Pydantic Version

2.6.1

### Python Version

3.11.8

**Comment by konstantinos1981** [comment]
It seems you are re-raising an exception that was already caught. Try modifying the code to handle the exception within the block

```python
async def redis_session() -> AsyncGenerator[Redis, None]:
    async with RedisPool() as client:
        try:
            yield client
        except Exception as e:
            logger.warning(f'redis_session error: {traceback.format_exc()}')
            raise e # Re-raise the caught exception
```

**Comment by YuriiMotov** [comment]
I tried to reproduce your issue, but I didn't succeed. Everything works fine.

Can you give a minimal working example that we can run and see that problem?

**Comment by YuriiMotov** [comment]
Closing this. Feel free to reopen it when you are ready to provide more info
